add_vendordep.py

```diff
@@ -0,0 +1,57 @@
+import json
+import argparse
+from pathlib import Path
+import shutil
+
+
+def add_vendordep(vendordep_filename, is_beta):
+    vendordep_contents = json.loads(vendordep_filename.read_bytes())
+    year = vendordep_contents['frcYear'] + "beta" if is_beta else ""
+
+    year_filename = Path(f"{year}.json")
+    year_contents = json.loads(year_filename.read_bytes())
+    
+    metadata_filename = Path(f"{year}_metadata.json")
+    metadata_contents = json.loads(metadata_filename.read_bytes())
+
+    for metadata_lib in metadata_contents:
+        if metadata_lib["uuid"] == vendordep_contents["uuid"]:
+            break
+    else:
+        raise Exception("This appears to be a new library that does not have metadata associated with it. Can not automatically update")
+
+    vendordep_destination = Path(f"{year}/{vendordep_filename.name}")
+    
+    year_contents.append(dict(
+        path=str(vendordep_destination),
+        name=metadata_lib["name"],
+        version=vendordep_contents["version"],
+        uuid=metadata_lib["uuid"],
+        description=metadata_lib["description"],
+        website=metadata_lib["website"],
+    ))
```

Comment:
YEAR.json is generated automatically now, adding the json to the directory is all that's necessary

Reply:
Where does this happen? If that happens I feel like there should be a gen-is-the-same test or these files can be removed

Reply:
Ok so it happens in `generate_year_bundles`, which only publishes to artifactory. The file in the root still has to be updated (either by hand or through a script) for the tests to pass.

I think either these files should be generated and checked, or removed. `generate_year_bundles` seems to enforce some of the checks the tests do so it might be redundant

Reply:
They will be removed, but removing them will break beta 1 & 2 VS Code extensions.

Reply:
We've already broken them. I'm working on removing all the obsolete stuff and documenting the new process